kafka-connect/kafka-connect/src/test/java/org/apache/iceberg/connect/data/RecordConverterTest.java

```diff
@@ -84,11 +93,18 @@
 import org.apache.kafka.connect.storage.ConverterType;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-public class RecordConverterTest {
+@ExtendWith(ParameterizedTestExtension.class)
+public class RecordConverterTest extends BaseWriterTest {
```

Comment:
I was hoping we'd keep this test specific to the conversion functions, and keep writer tests separate. Do you have thoughts on that?

Reply:
Was thinking in lines of conversion functions are no longer format agnostic as we are adding format info into deciding the record conversion, hence though it would be fair to test this E2E here. 

please let me know your thoughts considering above.

Reply:
Maybe we can create a dedicated test class for writer ?

Reply:
Shouldn't we have this parameterized by file type? The tests here make sense to me but I am only looking at this module for the first time for this PR.

Reply:
Ah iI see, the format only comes into play for UUID so the other parameterizations are essentially no-ops. Perhaps we just need one specalized test then ' testParquetUUIDSerialization"

Reply:
Sure, added this test, Thanks for suggesting this ! 
I also do think we need an E2E with write I can take this test as a followup to this pr as it would require refactoring of the writer tests.